test/unit/utils/install_context_test.rb

```diff
@@ -0,0 +1,86 @@
+require "helper"
+require "inspec/globals"
+require "inspec/utils/install_context"
+
+def check_install_contexts(test_expected_to_be_true, also_rubygem)
+  should_be_false = %w{chef-workstation chefdk docker
+                     habitat omnibus rubygem source}
+  should_be_false -= [test_expected_to_be_true]
+  should_be_false = should_be_false.map { |m| "#{m}_install?".tr("-", "_").to_sym }
+  should_be_false -= [:rubygem_install?] if also_rubygem
+  should_be_false.each { |m| expect(Inspec.method(m).call).must_equal false }
+
+  should_be_true   = ["#{test_expected_to_be_true}_install?".tr("-", "_").to_sym]
+  should_be_true  += [:rubygem_install?] if also_rubygem
+  should_be_true.each { |m| expect(Inspec.method(m).call).must_equal true }
+
+  expect(Inspec.guess_install_context).must_equal test_expected_to_be_true
+end
+
+describe Inspec::InstallContextHelpers do
+  describe "when it looks like a Docker installation" do
+    before do
+      Inspec.expects(:src_root).at_least_once.returns("/somewhere/gems/inspec-4.18.39")
+      File.expects(:exist?).at_least_once.with("/etc/alpine-release").returns(true)
```

Comment:
One more thing... this design means the tests can't be run in parallel because they're mocking on the globals... but since this is a module, this need not be the case at all.

```ruby
class TestThingy
  include Inspec::InstallContextHelpers
  attr_accessor :src_root
  # ...
end

obj = TestThingy.new src_root: "/somewhere/gems/inspec-4.18.39"
check_install_contexts obj, "docker", true
```

etc

This redesign make it easy to use `parallelize_me!` and not jam up any other tests that use `File`, `Directory` or other `Inspec` globals.

Reply:
`at_least_once` is required here by `mocha`; without it you are limited to exactly one invocation. I don't care how many times these are invoked.

Reply:
Re: the parallelization issue, I see how that can be addressed for src_root, but for File and Dir and I don't see how we can avoid mocking them.

Reply:
OK, I've put in a helper method to check for a path, and to look for paths in an optional ivar which will be set if it is under the test harness. That could probably be more graceful.